### PR TITLE
Add 'show' and 'clear' command for PG drop

### PR DESCRIPTION
**Pull request by ayurkiv-nvda**
#### - What I did
Added new CLI commands to view and clear PG dropped packet statistics.
Added the new CLI commands to the command reference guide.

#### - How I did it
I have added script which can generate table that shows current PG dropped packet statistic and also can clean it
Clean and Show is implemented with using serialized stats from previous execution.
To calculate current stats we do next step:
- take old serialized data (if no 'clear' executed all serialized data is 0)
- take current stat
- show diff between current and old

#### How to verify it
run
**show priority-group drop counters** - show current stat
**sonic-clear priority-group drop counters** - clear current stat

#### Previous command output (if the output of a command-line utility has changed)
N/A

#### New command output (if the output of a command-line utility has changed)

1) **show priority-group drop counters**
```
Ingress PG dropped packets:
     Port    PG0    PG1    PG2    PG3    PG4    PG5    PG6    PG7
---------  -----  -----  -----  -----  -----  -----  -----  -----
Ethernet0    800    801    802    803    804    805    806    807
Ethernet4    400    401    402    403    404    405    406    407
Ethernet8    100    101    102    103    104    105    106    107
...
```
2) **sonic-clear priority-group drop counters** 
Cleared PG drop counters

### CODE REVIEW

**Comment by neethajohn** [comment]
Please add unit tests

**Comment by liat-grozovik** [comment]
@ayurkiv-nvda please add to the PR description the output of the show command as part of
'#### New command output (if the output of a command-line utility has changed)'

File: doc/Command-Reference.md [hunk 1]
```diff
@@ -6094,11 +6094,14 @@ This command displays the user watermark for the queues (Egress shared pool occu
 
 **show priority-group**
 
-This command displays the user watermark or persistent-watermark for the Ingress "headroom" or "shared pool occupancy" per priority-group for  all ports
+This command displays:
+1) the user watermark or persistent-watermark for the Ingress "headroom" or "shared pool occupancy" per priority-group for  all ports.
```
Comment (liat-grozovik) [inline doc/Command-Reference.md:6098]: replate to 'The'
remove the extra space after for

File: scripts/pg-drop [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        except IOError as e:
+            print(e)
+            sys.exit(e.errno)
+        print("Cleared PG drop counter")
+
+def main():
+    parser = argparse.ArgumentParser(description='Display PG drop counter',
+                                     formatter_class=argparse.RawTextHelpFormatter,
+                                     epilog="""
+Examples:
```
Comment (liat-grozovik) [inline scripts/pg-drop:219]: if the 'Examples:' is part of the ArgumentParser why it is not aligned?

File: scripts/pg-drop [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+import os
+import sys
+from collections import OrderedDict
+
+from swsscommon.swsscommon import SonicV2Connector
+from natsort import natsorted
+from tabulate import tabulate
+
+# mock the redis for unit test purposes #
+try:
```
Comment (liat-grozovik) [inline scripts/pg-drop:20]: where is this going to be used? i don't see unittesting. need to add.

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

File: scripts/pg-drop (hunk 2 above)
Comment (ayurkiv-nvda) [reply scripts/pg-drop:219]: I copied style from other scripts:
https://github.com/Azure/sonic-utilities/blob/master/scripts/watermarkstat#L271
https://github.com/Azure/sonic-utilities/blob/master/scripts/portstat#L450
Should I align examples or leave as is ?

Comment (liat-grozovik) [reply scripts/pg-drop:219]: dont have a strong opinion. it was just strange for me. if you will have additional changes lets do, if not keep

**Comment by ayurkiv-nvda** [comment]
> This pull request **introduces 2 alerts** when merging [7ec47e2](https://github.com/Azure/sonic-utilities/commit/7ec47e22e3bc33227108ad33ff9086e4c9056090) into [4e27ad7](https://github.com/Azure/sonic-utilities/commit/4e27ad78ff3a82d75fd3fc3719d8cfe2bb9d3d8b) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-utilities/rev/pr-0d92baa99e407f67df6a9b00c5c546b5f9a90e88)
> 
> **new alerts:**
> 
> * 2 for Unused import

fixed

**Comment by ayurkiv-nvda** [comment]
> @ayurkiv-nvda please add to the PR description the output of the show command as part of
> '#### New command output (if the output of a command-line utility has changed)'

added

**Comment by ayurkiv-nvda** [comment]
> Please add unit tests

Done

File: clear/main.py [hunk 4]
```diff
@@ -216,6 +216,17 @@ def clear_wm_pg_shared():
     command = 'watermarkstat -c -t pg_shared'
     run_command(command)
 
+@priority_group.group()
+def drop():
```
Comment (neethajohn) [inline clear/main.py:220]: Please add the check so that clear can be executed only by 'root' . check the watermark group for reference

Comment (ayurkiv-nvda) [reply clear/main.py:220]: There is one problem:
For executing **sonic-clear priority-group drop counters** we use **runner.invoke**. But  there is no ability to run it with 'root' privileges  (I could not find appropriate solution). In tests that related to watermarks  there are no tests with  checking "clear counters" functionality.  
Possible solution:
- accept it problem (add **if os.geteuid() != 0** check) and add checking to "Root privileges are required for this operation" message  (can be temporary solution)
- do not add checking  **if os.geteuid() != 0** and leave as is
- try to find another way to run  command **sonic-clear priority-group drop counters** with root privilege 

Currently first proposal done locally

Comment (neethajohn) [reply clear/main.py:220]: Can you use a similar solution to bypass the root check for unit tests?
https://github.com/Azure/sonic-utilities/blob/master/scripts/ecnconfig#L94

**Review by neethajohn — APPROVED** [review]

**Comment by stephenxs** [comment]
Need to put the newly added `scripts/pg-drop` to `setup.py` as well. Fixed in #1586

**Pull request merged**